api/routes/model.js

```diff
@@ -0,0 +1,101 @@
+const express = require('express');
+const router = express.Router();
+var validate = require('express-jsonschema').validate;
+
+
+var ModelSchema = {
+    id: '/ModelSchema',
+    type: 'object',
+    properties: {
+      name: {
+        type: 'string',
+        required: true
+      },
+      sdg: {
+        type: 'array',
+        items:{type: 'integer'},
+        required: true
+      },
+      description: {
+        type: 'string',
+        required: true
+      },
+      stages: {
+        type: 'object',
+        patternProperties: {
+          ".*": {
+            type: 'object',
+            properties: {
+              stakeholders: {
+                type: 'array',
+                items: {type: 'string'}
+              },
+              challenges: {
+                type: 'array',
+                items: {type: 'string'}
+              },
+              insights: {
+                type: 'array',
+                items: {type: 'string'}
+              },
+              description: {
+                type: 'string'
+              }
+            }
+          }
+        }
+      }
+    }
+  };
+
+  router.get('/', function(req, res, next) {
+    const db = req.db;
+    const collection = db.get('modelCollection');
+    collection.find({},{},function(e,docs){
+        res.send(docs);
+    });
+  });
+  // TODO; Check Validate 
+  router.post('/', validate({body: ModelSchema}), function(req, res, next) {
+    const db = req.db;
+    const collection = db.get('modelCollection');
+    const data = req.body;
+
+    // Check if data includes proper fields
+
+    if (data) {
+        collection.insert(data, function(err, obj) {});
```

Comment:
how would error handling work for this? not a rhetorical question, actually wondering

Reply:
oh i mention in like the `function(err, obj)` stuff